crates/assistant/src/inline_assistant.rs

```diff
@@ -107,7 +107,7 @@ impl InlineAssistant {
             confirmed_assists: HashMap::default(),
             prompt_history: VecDeque::default(),
             prompt_builder,
-            telemetry: Some(telemetry),
+            telemetry,
```

Comment:
This `InlineAssistant` struct originally held an `Option<Arc<Telemetry>>`, however, in the `new()`, telemetry was always passed in as a non-option and then forced into an option via `Some(telemetry)`. I removed this option so that code within this module didn't have to try to account for it not existing.